pkg/watchdog/watchdog.go

```diff
@@ -98,7 +98,9 @@ const (
 
 // Watchdog holds the descriptor of an open watchdog driver.
 type Watchdog struct {
-	f *os.File
+	*os.File
+	unixSyscall        func(uintptr, uintptr, uintptr, uintptr) (uintptr, uintptr, unix.Errno)
```

Comment:
These should not be in this type. Consider defining an fs interface as in Andrew Gerrand's short presentation: https://talks.golang.org/2012/10things.slide#8. But we definitely don't want to drop things like this into this type. But this is a lot closer. 

Let's focus on this one package, and work hard to get the right pattern for these tests, and then revisit the other PRs in light of what we learn here. We're very close but it's critical that we get this pattern established now, so that we don't write lots of code that we have to maintain later.

Reply:
I see your point, no buts, only agreement.

Reply:
I did the same thing in the spidev package here: https://github.com/u-root/u-root/blob/master/pkg/spidev/spidev_linux.go#L179, so this would have to change too (in a separate PR). We need to work out what the best way to mock out a syscall is.

How about this?

```
type syscalls interface {
	syscall(uintptr, uintptr, uintptr, uintptr) (uintptr, uintptr, unixErrno)
	ioctlGetUint32(int, uint) (uint32, error)
}

type realSyscalls struct {}
func (realSyscalls) syscall(a, b, c, d uintptr) (uintptr, uintptr, unixErrno) {
	return unix.Syscall(a, b, c, d)
}
func (realSyscalls) ioctlGetUint32(a int, b uint) (uint32, error) {
	return ioctlGetUint32(a, b)
}

type Watchdog struct {
	*os.File
	syscalls
}
```

Reply:
Fine by me.